lib/api/WP_Auth0_Api_Client_Credentials.php

```diff
@@ -67,23 +67,23 @@ protected function handle_response( $method ) {
 			return self::RETURN_ON_FAILURE;
 		}
 
-		$response_body = json_decode( $this->response_body );
+		$response_body = json_decode( $this->response_body, true );
 
 		// If we have no access token, something went wrong upstream.
-		if ( empty( $response_body->access_token ) ) {
+		if ( ! isset( $response_body['access_token'] ) ) {
 			WP_Auth0_ErrorLog::insert_error( $method, __( 'No access_token returned.', 'wp-auth0' ) );
 			return self::RETURN_ON_FAILURE;
 		}
 
 		// Set the transient to expire 1 minute before the token does.
-		$expires_in  = ! empty( $response_body->expires_in ) ? absint( $response_body->expires_in ) : HOUR_IN_SECONDS;
+		$expires_in  = ! isset( $response_body['expires_in'] ) ? absint( $response_body['expires_in'] ) : HOUR_IN_SECONDS;
```

Comment:
I think the conditions were reversed here...

! empty --> not empty === a value is set

! isset --> no value is set === the value is empty

Reply:
Right you are, good catch! Committing a fix.